presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -349,59 +347,14 @@ public Scope getScope(Node node)
             return Optional.of(scopes.get(node));
         }
 
-        if (root == null) {
-            return Optional.empty();
-        }
-
-        GetScopeVisitor visitor = new GetScopeVisitor(scopes, node);
```

Comment:
Cmt msg could contain hint that it was not correct. "basically behaved like search in scopes map" doesn't convey this information for me.